terminology.md

```diff
@@ -261,17 +261,26 @@ A realm consists of a set of [intrinsic](#Intrinsic) objects, an ECMAScript glob
 
 ### Intrinsic
 #### Definition
-Built-in objects that are explicitly referenced by the algorithms of the ECMA262 specification.
+Built-in values that are required by the ECMA262 specification.
+Where observable (e.g., as [primordial](#Primordial)), intrinsic objects are realm-specific while intrinsic symbols are shared by all realms.
+The specification itself references "well-known" intrinsics with special notation (%&lt;name>% for objects; @@&lt;name> for symbols).
 #### Example
-`[]`
+%ForInIteratorPrototype% is the prototype of internal iterators that can be used to implement **`for (`** _Identifier_ **`in`** _Expression_ **`)`** statements
 #### Reference
-[ECMA262 Spec](https://tc39.es/ecma262/#sec-well-known-intrinsic-objects)
+[ECMA262 (objects)](https://tc39.es/ecma262/#sec-well-known-intrinsic-objects),
+[ECMA262 (symbols)](https://tc39.es/ecma262/#sec-well-known-symbols)
 
 ### Primordial
 #### Definition
-Global, mutable objects that must exist before any code runs.
+The subset of [intrinsic](#Intrinsic) values that are accessible to ECMAScript code and required to exist before any ECMAScript code runs.
 #### Example
-`%ArrayPrototype%`
+%Array%, the initial value of a realm's Array constructor, is accessible as `Array`.
+
+### Undeniable
+#### Definition
+The subset of [primordial](#Primordial) values that are accessible to ECMAScript code without reference to named bindings (i.e., by means of syntax alone).
+#### Example
+%Array.prototype%, the initial value of a realm's Array prototype is accessible as `[].__proto__`.
```

Comment:
This is fudging a bit; `[].__proto__` isn't guaranteed to return %Array.prototype% if Object.prototype.__proto__ has been removed or replaced. Is there any primordial _truly_ undeniable? And if not, should "Undeniable" be removed, renamed, or updated to incorporate further assumptions?

Reply:
If you delete `constructor` and `__proto__` from every builtin prototype, and you replace all the appropriate reflection functions:
 - `Object.getOwnPropertyDescriptor`
 - `Object.getOwnPropertyDescriptors`
 - `Reflect.getOwnPropertyDescriptor`
 - `Object.getPrototypeOf`
 - `Reflect.getPrototypeOf`

then all primordials *in the current realm* may indeed be deniable; if you also replace any environment-specific means of creating an unmodified realm, then that may extend to all primordials in any realm.

(I reviewed [test262's intrinsic list](https://github.com/tc39/test262/blob/a1acc23cd85c552cde9c06cfed300c58d4b5b04c/harness/wellKnownIntrinsicObjects.js) and [es-abstract's intrinsic list](https://github.com/ljharb/es-abstract/blob/3641bbb4196ad6bc31d443bd2fbdb81c61fb664e/GetIntrinsic.js#L53) to confirm my thinking on this)

@erights, thoughts?

Reply:
This seems technically correct to me. But these means of denial are so extreme that we should never expect them to be used in practice. We should explain this qualifier on the term "undeniable" but keep the term for anything that can be reached by "syntax plus ascending the prototype chain". If you like, "effectively undeniable" would be technically correct.

Reply:
Indeed, if you preserve the ability to obtain getters, and to get the [[Prototype]], then there exists a great many undeniables.